### PR TITLE
bsc#1174919

### PR DESCRIPTION
**Pull request by bear454**
Deploy in EKS can fail due to operations taking longer than the expiration period of the k8s auth token generated by terraform, and terraform not having a built-in mechanism to renew the token:

https://github.com/hashicorp/terraform/issues/24886

I'm working around this by ensuring kubernetes operations occur up-front, and helm operations use the `kubecontrol` file instead, which request a new token on each use.

In addition, due to other outstanding issues, we've elected to upgrade the `helm` and `kubernetes` provider versions.

### CODE REVIEW

File: eks/modules/services/wait_for_url.sh [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
     echo "connecting..."
     code=$(curl -k -o /dev/null -Isw '%{http_code}\n' --connect-timeout 5 --max-time 5 $URL)
-    if [[ "$code" == '200' ]]; then
+    if [ "$code" -eq 200 ]; then
         break;
     fi
-    echo "...but received status ${code}; will check again in a minute ..."
-    sleep 60
+    echo "...but received status $code; will check again shortly ..."
+    sleep 10
```
Comment (satadruroy) [inline eks/modules/services/wait_for_url.sh:12]: UAA doesn't come up until about 20+ minutes into the CAP deploy and because of the asynchronous nature of the helm installs this script can kick in right after the helm installs have triggered so waiting for only 10s is lot of unnecessary polling but only gaining little less than a minute in triggering the final metrics install.

File: eks/modules/services/scfcharts.tf [hunk 2]
```diff
@@ -185,29 +179,35 @@ resource "null_resource" "wait_for_uaa" {
...
   ]
 
-  provider   = helm.metrics
   name       = "susecf-metrics"
   repository = "https://kubernetes-charts.suse.com"
   chart      = "metrics"
   version    = "1.2.1"
   namespace  = "metrics"
-  wait       = "false"
+  wait       = "true"
```
Comment (satadruroy) [inline eks/modules/services/scfcharts.tf:210]: We don't need this to be true - there's nothing else dependent on it. The CAP deployment will take its time too and we are telling the end user to check the pod status for readiness.

File: eks/modules/services/scfcharts.tf [hunk 3]
```diff
@@ -127,7 +119,9 @@ resource "helm_release" "scf" {
 resource "helm_release" "stratos" {
   depends_on = [
     helm_release.scf,
-    kubernetes_namespace.stratos
+    kubernetes_namespace.stratos,
+    null_resource.cluster_issuer_setup,
```
Comment (satadruroy) [inline eks/modules/services/scfcharts.tf:123]: Why is this needed? In v1, stratos is not using cert-manager.

File: eks/modules/services/helm.tf [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
+      args = [
+	      "token",
+        "-i",
+        var.aws-eks-cluster-name
+      ]
+      env = {
+        "AWS_ACCESS_KEY_ID" = var.access_key_id
+        "AWS_SECRET_ACCESS_KEY" = var.secret_access_key
+      }
+    }
```
Comment (satadruroy) [inline eks/modules/services/helm.tf:20]: Is this to avoid the token expiry problem? Could using the `load_config_file = true` work?

**Review by satadruroy — APPROVED** [review]
Mostly minor comments and approving based on our discussion but we should look at some of the suggested cleanup.

**Comment by bear454** [comment]
Issues noted; I'll resolve in a subsequent PR.
